cxx-sensors/src/main/java/org/sonar/cxx/sensors/compiler/gcc/CxxCompilerGccParser.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Sonar C++ Plugin (Community)
+ * Copyright (C) 2010-2018 SonarOpenCommunity
+ * http://github.com/SonarOpenCommunity/sonar-cxx
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.cxx.sensors.compiler.gcc;
+
+import java.io.File;
+import java.util.List;
+import java.util.Scanner;
+import java.util.regex.MatchResult;
+import java.util.regex.Pattern;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.cxx.sensors.compiler.CompilerParser;
+
+/**
+ * {@inheritDoc}
+ */
+public class CxxCompilerGccParser implements CompilerParser {
+
+  private static final Logger LOG = Loggers.get(CxxCompilerGccParser.class);
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void processReport(final SensorContext context, File report, String charset, String reportRegEx, List<Warning> warnings)
+    throws java.io.FileNotFoundException {
+
+    LOG.info("Parsing 'GCC' format ({})", charset);
+
+    try (Scanner scanner = new Scanner(report, charset)) {
+      Pattern p = Pattern.compile(reportRegEx, Pattern.MULTILINE);
+      LOG.debug("Using pattern : '{}'", p);
+      MatchResult matchres;
+      while (scanner.findWithinHorizon(p, 0) != null) {
+        matchres = scanner.match();
+        String filename = matchres.group(1).trim();
```

Comment:
In case group not available could be null

Reply:
Since findWithinHorizon() is used as pre-requirement for match(), the MatchResult contains the proper results. Maybe we could assert the number of groups at runtime (matchres.groupCount() >=5). However it's sufficient to double-check the corresponding expression IMHO